monitoring/nethealth.go

```diff
@@ -226,11 +235,28 @@ func (c *nethealthChecker) setTimeoutSeries(name string, series []int64) error {
 	return c.timeoutStats.Set(name, series, timeoutStatsTTLSeconds)
 }
 
-// fetchMetrics collects the network metrics from the nethealth service.
-// Metrics are returned as an array of bytes.
-func fetchMetrics(ctx context.Context, ip string, port int) ([]byte, error) {
-	addr := fmt.Sprintf("http://%s:%d", ip, port)
+// getNethealthAddr returns the address of the local nethealth pod.
+func (c *nethealthChecker) getNethealthAddr() (string, error) {
+	pods, err := c.Client.CoreV1().Pods(monitoringNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
 
+	// Find nethealth pod with matching host ip address.
+	for _, pod := range pods.Items {
+		if !strings.Contains(pod.GetName(), nethealthName) {
```

Comment:
You should list pods using specific label/field selectors to avoid this - otherwise you're running the risk of picking something arbitrary.

Reply:
Addressed in f8d89e9.
Does this is implementation look correct?